test/e2e/specs/scaleupdown.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+)
+
+var _ = Describe("Scale Up/Down E2E tests [ScaleUpDown] [Real] [LongRunning] [EveryPR]", func() {
+	var (
+		azurecli *azure.Client
+		occli    *openshift.Client
+		manifest = flag.String("manifest", "../../_data/manifest.yaml", "Path to the manifest to send to the RP")
+	)
+
+	const (
+		sampleDeployment = "hello-openshift"
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+		occli, err = openshift.NewCustomerAdminClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be possible to maintain a healthy cluster after scaling it out and in", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Scaling out the compute scale set to 2 nodes")
+		By("Parsing the external manifest")
+		external, err := parseExternalConfig(*manifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By(fmt.Sprintf("Updating the size of the compute scale set to %d in the external manifest", 2))
+		Expect(*external.Properties.AgentPoolProfiles[1].Name).To(Equal("compute"))
+		*external.Properties.AgentPoolProfiles[1].Count = 2
+
+		By("Calling update on the rp with the updated external manifest")
+		pluginConfig, err := fakerp.GetPluginConfig()
+		Expect(err).NotTo(HaveOccurred())
+		updated, err := azurecli.UpdateOSACluster(ctx, external, pluginConfig)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(updated).NotTo(BeNil())
+
+		By("Creating a namespace as cluster admin")
+		namespace, err := randomstring.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 5)
+		Expect(err).ToNot(HaveOccurred())
+		namespace = "e2e-test-" + namespace
+		err = occli.CreateProject(namespace)
+		Expect(err).ToNot(HaveOccurred())
+		defer func() {
+			err := occli.CleanupProject(namespace)
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		By("Creating a 2-replica sample deployment with pod anti-affinity to ensure pods are spread across nodes")
+		int32Ptr := func(i int32) *int32 { return &i }
+		deployment := &apiappsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      sampleDeployment,
+				Namespace: namespace,
+			},
+			Spec: apiappsv1.DeploymentSpec{
+				Replicas: int32Ptr(int32(2)),
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": sampleDeployment,
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: sampleDeployment,
+						Labels: map[string]string{
+							"app": sampleDeployment,
+						},
+					},
+					Spec: corev1.PodSpec{
+						Affinity: &corev1.Affinity{
+							PodAntiAffinity: &corev1.PodAntiAffinity{
```

Comment:
Pretty sure you don't need this, [the scheduler uses the label selector speading predicate by default](https://github.com/openshift/openshift-azure/blob/0d8751f857bc7b8777c9e1a15fcb859895a762bc/pkg/arm/data/master-startup.sh#L385).

Reply:
hey thats good to know. I'll test it out and take this section out if tests are fine